packages/components/src/InputPhoneNumber/InputPhoneNumber.stories.mdx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { ExampleWithHooks } from "mdxUtils/ExampleWithHooks";
+import { InputPhoneNumber } from "@jobber/components/InputPhoneNumber";
+
+<Meta title="Components/InputPhoneNumber" component={InputPhoneNumber} />
+
+# InputPhoneNumber
+
+Automatically format your user's phone number as they type.
+
+```ts
+import { InputPhoneNumber } from "@jobber/components/InputPhoneNumber";
+```
+
+<Canvas>
+  <Story
+    name="InputPhoneNumber"
+    args={{
+      placeholder: "Phone number",
+    }}
+  >
+    {args => {
+      const [value, setValue] = useState(args.value);
+      return <InputPhoneNumber {...args} value={value} onChange={setValue} />;
+    }}
+  </Story>
+</Canvas>
+
+<ArgsTable of={InputPhoneNumber} story="InputPhoneNumber" />
+
+## Usage guidelines
+
+Due to the auto-format, this component is restricted to only be used as a
+controlled component. That means the `value` and `onChange` prop is required,
+unlike the other input components.
+
+## With country code as a prefix
+
+Keep in mind that the country code won't be added to the value that the
+component would return. It is simply for visual purposes.
+
+<Canvas>
+  <ExampleWithHooks>
+    {() => {
+      const [value, setValue] = useState("");
+      return (
+        <InputPhoneNumber
+          placeholder="Phone number"
+          prefix={{ label: "+1" }}
+          value={value}
+          onChange={setValue}
+        />
+      );
+    }}
+  </ExampleWithHooks>
+</Canvas>
+
+## Dealing with existing data
+
```

Comment:
I wonder if we should allow the user an implicit kind of "revert" behaviour if they start editing the field when they didn't mean to.
Here's a use case I am imagining. The field has some existing data ("555-TUNA") and they accidentally edit the field (or could this even happen if they tab through?)... Currently, we would format the data with the best guess we have, but they have lost the original data. If they blurred the field, could we revert to their existing invalid data?
Maybe that makes sense after we add validation? I.e. if you first interact with the field when it contains existing, invalid data, and you edit it but don't produce valid data.

Hmm, on second thought, are we wanting to require them to enter a valid number? Is the intent of this field to only be used in contexts where we enforce "valid" inputs? 

Going to drop this note here, but happy to discuss. I could see a potential future where we want formatting but not strict validation. But maybe not yet... ?

Reply:
Yeah, left it as is since it will be used on newer features! Older features tho, it will be a bit problematic when it suddenly changes and you have a non-number value in there.

It only autoformats it when you do try and change the value. Focus and blur won't trigger it.

> Hmm, on second thought, are we wanting to require them to enter a valid number? Is the intent of this field to only be used in contexts where we enforce "valid" inputs?

Yes. We want phone numbers that consists of numbers